blog/2024-11-effect-afterrendereffect/README.md

```diff
@@ -0,0 +1,504 @@
+In response, Angular's new approach now allows setting signals within `effect()` by default, removing the need for `allowSignalWrites`.
+This more flexible design reflects Angular's commitment to simplifying the development experience.
+See the [official blog post](https://blog.angular.dev/latest-updates-to-effect-in-angular-f2d2648defcd) that confirms this new guidance.
+
+We interpret this new information in the following way:
+> 💡 **It is now a valid case to use `effect()` for state updates or side effects that are difficult to achieve with other reactive primitives, such as `computed()`**.
+
+This change to the paradigm is in line with new features introduced in Angular 19, such as [`linkedSignal()`](https://angular.schule/blog/2024-11-linked-signal) and `resource()`.
+Both help to maintain cleaner and more declarative state management patterns where possible. 
+Good patterns are no longer enforced by the `allowSignalWrites` flag, but instead by useful high-level signal APIs.
+
+With this shift, here's a new general rule of thumb:
+- **Use `effect()`** for tasks traditionally performed in `ngOnInit` or `ngOnChanges`.
+- **Use `afterRenderEffect()`** for tasks traditionally handled in `ngAfterViewInit` or `ngAfterViewChecked`, or when you need to interact directly with rendered DOM elements.
+
+Let's dive into the specifics! 🚀
+
+
+
+## Core Differences Between `effect()` and `afterRenderEffect()`
+
+Both `effect()` and `afterRenderEffect()` are designed to track and respond to changes in signals, but they differ in timing and use cases.
+
+- **`effect()`** runs as part of the Angular change detection cycle and can now safely modify signals without any additional flags.
+- **`afterRenderEffect()`** is a lower-level API that executes after the DOM has been updated. 
+  It's particularly suited for tasks that require interacting directly with the DOM, such as measuring element sizes or making complex visual updates in phases.
+
+Here's a simple comparison to illustrate how these functions operate:
+
+```typescript
+counter = signal(0);
+
+effect(() => {
+  console.log(`Current counter value: ${this.counter()}`);
+});
+
+afterRenderEffect(() => {
+  console.log('DOM rendering completed for this component');
+});
+```
+
+As expected, the console output for `afterRenderEffect()` is triggered after the output of `effect()`.
+
+
+## Introducing `effect()`
+
+Angular supports two types of effects: **component effects** and **root effects**. 
+Component effects are initiated when `effect()` is called within a component, directive, or a service tied to them. 
+Root effects, on the other hand, are initiated when `effect()` is called outside the component tree, such as in a [singleton service](https://angular.dev/guide/ngmodules/singleton-services), or by setting the `forceRoot` option.
```

Comment:
Okay, I think it's clearer to me now: "service tied to a component" obviously means a service provided in the component's NodeInjector. [This code](https://github.com/angular/angular/blob/main/packages/core/src/render3/reactivity/effect.ts#L183-L194) says in the comment:

> This effect was created in the context of a view, and will be associated with the view.

When this is not the case, it will become a root effect.

- Effect in service provided in root injector: Root Effect
- Effect in service provided in component: View Effect
- Effect in component: View Effect

Alright, just wanted to make sure we have understood this at least superficially before we publish it 🙂

Reply:
I think we're on the same line, but it wasn't clear to me from just the text above. I think this text can also be removed because for us there is no practical difference between root and view effects. It's more confusing that there are two types that seem to be equal.

If you don't want to skip it completely, maybe let's talk about it directly?

> In this section we only talk about effects that are created from within a component. These are called **component effects** and allow to safely read and write component properties and signals.
> It is also possible to create effects in services. If the services is provided at root level in the application, these effects are called **root effects**. 


and then maybe the two sentences from below. This makes it clearer that we mostly talk about component effects and that the two are not equally used.